### PR TITLE
Flake8Bear: Work in progress

### PR DESCRIPTION
**Pull request by srisankethu**
Did not default flags like max-line-length, exclude, ignore. @sils1297 insisted on adding a PR for review. Need to add tests if bears is :+1:

### CODE REVIEW

**Comment by AbdealiJK** [comment]
review is at https://github.com/coala-analyzer/coala-bears/commit/b4fded46d2387aaee82f0de5283a27bad7005bf5

**Comment by srisankethu** [comment]
I tried importing flake8.main but in check_file() and check_code() are returning int values. So should I try without importing?

**Comment by AbdealiJK** [comment]
It shouldnt be returning an int I think ...
https://github.com/PyCQA/flake8/blob/master/flake8/main.py#L79 says it is a
very simple function.
Maybe we want to use
https://github.com/PyCQA/flake8/blob/master/flake8/engine.py#L230 directly
? it seems more powerful

**Comment by srisankethu** [comment]
param: ouput is NONE. I am not understanding, what's my mistake ?
https://github.com/coala-analyzer/coala-bears/pull/230/files#diff-4a6e8b080ef21dc8145e9386c1d06c3fR33

**Comment by Makman2** [comment]
@srisankethu if this is a WIP, could you close the PR temporarily, since if you change stuff our CI activates everytime, this blasts build times a bit when we have a coding "rush-hour" :D
If you are done you can easily reopen it again :+1: :)

**Comment by srisankethu** [comment]
@Makman2 Ok :+1: . Can you check what i might have missed ?

File: bears/python/Flake8Bear.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from coalib.bearlib.abstractions.Lint import Lint
+from coalib.bears.LocalBear import LocalBear
+
+if sys.platform.startswith('win'):
+    DEFAULT_CONFIG = os.path.expanduser(r'~\.flake8')
+else:
+    DEFAULT_CONFIG = os.path.join(
+        os.getenv('XDG_CONFIG_HOME') or os.path.expanduser('~/.config'),
+        'flake8'
+    )
```
Comment (Makman2) [inline bears/python/Flake8Bear.py]: please align the last bracket onto the line with the last element:
`'flake8')`

File: bears/python/Flake8Bear.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+class Flake8Bear(Lint, LocalBear):
+
+    """ Checks the code according to pep8 and pyflake8 standards """
+
+    def lint(self, filename, file):
+        self.gives_corrected = False
```
Comment (Makman2) [inline bears/python/Flake8Bear.py]: this one needs to be at class level

Comment (Makman2) [inline bears/python/Flake8Bear.py]: this one needs to be at class level

File: bears/python/Flake8Bear.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    DEFAULT_CONFIG = os.path.expanduser(r'~\.flake8')
+else:
+    DEFAULT_CONFIG = os.path.join(
+        os.getenv('XDG_CONFIG_HOME') or os.path.expanduser('~/.config'),
+        'flake8'
+    )
+
+
+class Flake8Bear(Lint, LocalBear):
+
```
Comment (Makman2) [inline bears/python/Flake8Bear.py]: no newline here :+1:

File: bears/python/Flake8Bear.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import sys
+import os
+
+from flake8.engine import get_style_guide
+
+from coalib.bearlib.abstractions.Lint import Lint
+from coalib.bears.LocalBear import LocalBear
+
+if sys.platform.startswith('win'):
```
Comment (Makman2) [inline bears/python/Flake8Bear.py]: Please use `platform.system()` to check for the OS

File: bears/python/Flake8Bear.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import sys
+import os
+
+from flake8.engine import get_style_guide
+
+from coalib.bearlib.abstractions.Lint import Lint
+from coalib.bears.LocalBear import LocalBear
+
```
Comment (Makman2) [inline bears/python/Flake8Bear.py]: please another newline here :)

File: bears/python/Flake8Bear.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        return self.process_output(list(output), filename, file)
+
+    def run(self,
+            filename,
+            file,
+            max_line_length: int=80,
+            ignore=(),
+            flake8_format='',
+            max_complexity=-1):
+
```
Comment (Makman2) [inline bears/python/Flake8Bear.py]: superfluos newline

**Comment by Makman2** [comment]
And yeah tests are missing :+1:

**Comment by srisankethu** [comment]
@Makman2 I will check the code-style part later! But the code isn't working :( I wanted to know what i am missing here, why is param: option is None

**Comment by Makman2** [comment]
@srisankethu what do you mean?

**Comment by srisankethu** [comment]
@Makman2 I meant that output is of type None when i am running it, but it shouldn't be the case

**Comment by Makman2** [comment]
I don't know how flake8 works, that's your duty to debug and find out what's the problem to get stuff running :)

**Comment by srisankethu** [comment]
@Makman2 Yes! I'm trying :+1:

**Comment by srisankethu** [comment]
In param: report is a object <flake8.reporter.QueueReport> . This class has a method get_file_results to print results and return an overall count(this method is of no use for us). I tried using it but returns QueueReport has no attribute _deferred_print(This must be a bug in flake8). @AbdealiJK I found no method that stores the error_messages.

**Comment by AbdealiJK** [comment]
@Makman2 _deferred_print is obviously not a bug
If you think thats a bug, please try cloning and testing it locally and check if any error arises.

The Reporters are inheriting from pep8 Reporters.
Which are doing something - https://github.com/PyCQA/pycodestyle/blob/c8e0911f5df9d587e16bccf70cd48cac05d9c49d/pep8.py#L1749

**Comment by srisankethu** [comment]
@AbdealiJK i was checking the code from a .tar.gz that i downloaded. There seems to be difference between these two packages.

**Pull request closed without merging**